docs/choreolib/auto-factory.md

```diff
@@ -0,0 +1,112 @@
+
+# AutoFactory
+
+ChoreoLib provides the `AutoFactory` class as a higher level API to make it easier to create complex auto routines inside your robot code.
+
+## Basic Usage
+
+You can set up the `AutoFactory` by calling `Choreo.createAutoFactory`.
+The `AutoFactory` is recommended to be created in the robot class constructor instead of the drive subsystem.
+
+```java
+// The most basic usage of the AutoFactory
+class Robot extends TimedRobot {
+  /** A swerve drive subsystem */
+  private final Drive drive = ...;
+  /** An object that manages information about robot position */
+  private final Localizer localizer = ...;
```

Comment:
Is this common? Do most teams structure their code like this? If so I'm fine with it but most teams I've seen will just use the drivebase to get their pose so I don't think this makes sense.

Reply:
Most code I've seen has the odometry/pose estimator in the drivetrain subsystem.

Reply:
Yeah agreed. I think it makes more sense to just use drive::getPose or something of the like